### ISSUE TITLE
Refactor the testsuite to be more fine grained in assignment of tests to workers

### ISSUE THREAD

**Issue by ViralBShah** [JuliaSparse/SparseArrays.jl#149]
The [test/sparse.jl](https://github.com/JuliaSparse/SparseArrays.jl/blob/main/test/sparse.jl) is a huge file. Splitting it into a few smaller files will possibly help alleviate https://github.com/JuliaLang/julia/issues/45754 and also leverage more processors during tests.

**Comment by ViralBShah** [comment]
I don't think so. Seems like the julia runtests will pick each file to run in its own worker. The modules probably do help to avoid accidental reuse of global variables and that sort of stuff.

**Comment by ViralBShah** [comment]
How about just get rid of the `testgroups` stuff since the solvers have to be manually included anyways? Also maybe move the multi-threaded execution test into `solvers/threads.jl`?

**Comment by ViralBShah** [comment]
I don't see testgroups files in most of the tests though, which made me wonder. I thought it would be a lot more common.

```
➜  julia git:(master) ✗ find . -name testgroups
./stdlib/Dates/test/testgroups
./stdlib/SparseArrays-2bbdd7a12ead8207593655c541ba347761a9c663/test/testgroups
./stdlib/LinearAlgebra/test/testgroups
./stdlib/LibGit2/test/testgroups
```

**Comment by DilumAluthge** [comment]
Yeah, most stdlibs don't have a testgroups file, but we probably should add them to more stdlibs.

We definitely want to have them in the linear algebra-related stdlibs, because those tests take so long.

**Comment by ViralBShah** [comment]
How to handle the [`USE_GPL_LIBS`](https://github.com/JuliaSparse/SparseArrays.jl/blob/2bbdd7a12ead8207593655c541ba347761a9c663/test/runtests.jl#L7) with `testgroups`?

**Comment by DilumAluthge** [comment]
Fatfinger

**Comment by DilumAluthge** [comment]
Hmmm. Presumably, USE_GPL_LIBS is true by default in vanilla build of Julia?

**Comment by ViralBShah** [comment]
It is - but we do need to support the flag and not assume it's always present. I suppose it is ok if the sparse solver tests all run in one thread.

**Comment by DilumAluthge** [comment]
I think you'll need to add an `if Base.USE_GPL_LIBS` guard to the top of each test file

**Comment by ViralBShah** [comment]
Fixed by #151 and #154
